paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLecternPageChangeScriptEvent.java

```diff
@@ -0,0 +1,75 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import io.papermc.paper.event.player.PlayerLecternPageChangeEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerLecternPageChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player flips lectern page
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when the player flips to a page in a lectern.
+    //
+    // @Context
+    // <context.book> returns an ItemTag of the book in the lectern.
+    // <context.lectern> returns a LocationTag of the lectern.
+    // <context.old_page> returns the last page the player was on. Page numbers follow zero-based numbering, starting at 0.
+    // <context.new_page> returns the new page that the player flipped to.
+    // <context.page_direction> returns the direction in which the player flips the lectern book page.
+    //
+    // @Player when the attacked entity is a player.
+    //
+    // -->
+
+    public PlayerLecternPageChangeScriptEvent() {
+        registerCouldMatcher("player flips lectern page");
+    }
+
+    public PlayerLecternPageChangeEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        return super.matches(path);
+    }

```

Comment:
No need to override it if it's just a super call anyway, although according to the meta there is a location, so should add the check for that here.
Might be handy to add a switch for the book as well, for easier use when you have custom book items and such.